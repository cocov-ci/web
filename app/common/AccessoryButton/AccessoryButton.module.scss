@import "styles/variables.module";

.base {
  background: $colorGrayDark;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #fff;
}

.round {
  border: 1px solid $colorAccessoryButtonRoundBorder;
  border-radius: 50%;
  width: 24px;
  height: 24px;
}

.squared {
  border-radius: 8px;
  height: 33px;
  width: 33px;
  border: 0;
}
