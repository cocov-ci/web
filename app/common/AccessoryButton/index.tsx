import classNames from 'classnames'
import Image from 'next/image'
import Link from 'next/link'
import React, { HTMLAttributeAnchorTarget } from 'react'

import Url from 'types/Url'

import styles from './AccessoryButton.module.scss'

type AccessoryButtonProps = {
  children?: React.ReactNode
  className?: string
  kind: 'squared' | 'round' | 'squared-muted'
  onClick?: (ev: React.MouseEvent) => void
  title?: string
  href?: Url
  disabled?: boolean
  hrefTarget?: HTMLAttributeAnchorTarget
}

const AccessoryButton = ({
  children,
  className,
  kind,
  onClick,
  title,
  href,
  hrefTarget,
  disabled,
}: AccessoryButtonProps) => {
  const baseProps = {
    className: classNames(
      styles.base,
      className,
      ...kind.split('-').map(i => styles[i]),
      {
        [styles.disabled]: disabled,
      },
    ),
    title: title,
  }

  if (href) {
    return (
      <Link href={href} target={hrefTarget}>
        <div {...baseProps}>{children}</div>
      </Link>
    )
  }

  return (
    <button {...baseProps} onClick={ev => onClick && onClick(ev)}>
      {children}
    </button>
  )
}

export const CopyIcon = () => (
  <Image alt="" height={13} src="/icons/copy.svg" width={13} />
)

export const CheckIcon = () => (
  <Image alt="" height={11} src="/icons/check.svg" width={16} />
)

export const HelpIcon = () => (
  <Image alt="" height={10.5} src="/icons/help.svg" width={5.83} />
)

export default AccessoryButton
