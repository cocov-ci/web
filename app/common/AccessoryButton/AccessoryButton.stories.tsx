import type { Meta, StoryObj } from '@storybook/react'

import AccessoryButton, { CopyIcon, HelpIcon } from './index'

const meta: Meta<typeof AccessoryButton> = {
  title: 'Common/AccessoryButton',
  component: AccessoryButton,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof AccessoryButton>

export const CopyButton: Story = {
  args: {
    kind: 'squared',
    children: CopyIcon,
  },
}

export const HelpButton: Story = {
  args: {
    kind: 'round',
    children: HelpIcon,
  },
}
