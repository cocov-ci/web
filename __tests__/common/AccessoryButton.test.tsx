import { fireEvent, render, screen } from '@testing-library/react'

import AccessoryButton, {
  CheckIcon,
  CopyIcon,
  HelpIcon,
} from 'app/common/AccessoryButton'

import '@testing-library/jest-dom'

describe('common/AccessoryButton', () => {
  const { container } = render(
    <AccessoryButton kind="round">
      <HelpIcon />
    </AccessoryButton>,
  )

  it('renders AccessoryButton', () => {
    expect(screen.getByRole('button')).toBeVisible()
  })

  it('renders AccessoryButton with <a>', () => {
    const { container } = render(
      <AccessoryButton href="/" kind="squared">
        <CopyIcon />
      </AccessoryButton>,
    )

    expect(container.querySelector('a')).toBeVisible()
  })

  it('triggers onClick event', () => {
    const handleClick = jest.fn()
    render(
      <AccessoryButton kind="squared" onClick={handleClick}>
        <CheckIcon />
      </AccessoryButton>,
    )
    fireEvent.click(screen.getByRole('button'))
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('renders AccessoryButton snapshots', () => {
    expect(container).toMatchSnapshot()
  })
})
