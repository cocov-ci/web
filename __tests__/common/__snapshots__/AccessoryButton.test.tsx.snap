// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/AccessoryButton renders AccessoryButton snapshots 1`] = `<div />`;
